lib/bolt/transport/local.rb

```diff
@@ -146,25 +154,32 @@ def run_task(target, task, arguments, _options = {})
               environment_params = ""
             end
 
-            output =
-              if Powershell.powershell_file?(script) && stdin.nil?
-                command = Powershell.run_ps_task(arguments, script, input_method)
-                command = environment_params + Powershell.shell_init + command
+            if Powershell.powershell_file?(script) && stdin.nil?
+              command = Powershell.run_ps_task(arguments, script, input_method)
+              command = environment_params + Powershell.shell_init + command
+              interpreter ||= 'powershell.exe'
+              output =
                 if input_method == 'powershell'
-                  @conn.execute(command, dir: dir, env: "powershell.exe")
+                  @conn.execute(command, dir: dir, interpreter: interpreter)
```

Comment:
Hooray! I never liked that we used `env` for that before.

Reply:
Yeah that was not great. For some reason at the time I was hesitant to make any changes to `shell.rb`.